internal/git/gitsync_processor.go

```diff
@@ -1,19 +1,76 @@
 package git
 
 import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	numaplanenumaprojiov1 "github.com/numaproj-labs/numaplane/api/v1"
+	v1 "github.com/numaproj-labs/numaplane/api/v1"
 )
 
+const messageChanLength = 5
+
+type Message struct {
+	Updates bool
+	Err     error
+}
+
 type GitSyncProcessor struct {
+	gitSync     v1.GitSync
+	channels    map[string]chan Message
+	k8Client    client.Client
+	clusterName string
+}
+
+func isValidTag(tag string) bool {
+	match, _ := regexp.MatchString(`^[vV]?\d+\.\d+\.\d+$`, tag)
+	return match
 }
 
-func NewGitSyncProcessor(gitSync *numaplanenumaprojiov1.GitSync, k8sClient client.Client) (*GitSyncProcessor, error) {
-	return &GitSyncProcessor{}, nil
+func cloneRepository(repoUrl string, reference string) (*git.Repository, error) {
```

Comment:
can you add a comment for the fact that we support tag, branch, and commit hash? (see my other comment below regarding commit hash)